synapse/util/retryutils.py

```diff
@@ -215,6 +223,12 @@ def __exit__(
             self.failure_ts = None
             retry_last_ts = 0
             self.retry_interval = 0
+        elif valid_err_code:
+            # We got a potentially valid error code back. We don't reset the
+            # timers though, as the other side might actually be down anyway
+            # (e.g. some deprovisioned servers will always return a 404 or 403,
```

Comment:
Aside: would be nice if decommissioned servers could send us `410 Gone` so that we would never bother to retry sending to them (unless they send us traffic in the future).

Reply:
> Aside: would be nice if decommissioned servers could send us `410 Gone` so that we would never bother to retry sending to them (unless they send us traffic in the future).

Now this would be awesome best "workaround" for servers that were "pulled the plug unsafely" and have no means/desires to build matrix server again but wants to be good citizen after the fact. Simple idea, helps what it can.